rasa/core/test.py

```diff
@@ -651,9 +617,14 @@ def _collect_action_executed_predictions(
                 prediction.max_confidence,
                 event.timestamp,
                 metadata=prediction.action_metadata,
+                predicted_action_unlikely_intent=previously_predicted_action_unlikely_intent,
             )
         )
-        if fail_on_prediction_errors and has_prediction_target_mismatch:
+        if (
+            fail_on_prediction_errors
+            and predicted_action != ACTION_UNLIKELY_INTENT_NAME
+            and expected_action != ACTION_UNLIKELY_INTENT_NAME
```

Comment:
Doesn't this break the requirement we fail if the test story contains `ACTION_UNLIKELY_INTENT_NAME` but we don't predict it?

Reply:
Updated the condition and added a test for it (`test_action_unlikely_intent_fail_on_prediction_errors`)